Adds flags to generate OSL only for specified mx files and types

amartinez-cg opened the pull request:
-s, --shader specifies a single mx file to generate osl for, e.g. mx_absval
-t, --types specifies a space-separated list of types to generate for, e.g. FLOAT COLOR COLOR4

derekhaase commented on src/build-scripts/build_materialX_osl.py:
```diff
@@ -210,9 +210,13 @@ def main():
     parser.add_argument('-compile', '--compile', default=0, help='Compile generated osl files 0|1.  Default: 0')
-    
+    parser.add_argument('-s', '--shader', default='', help='Specify a single mx shader to convert without the file extension, e.g. mx_add.  Default: none')
+    parser.add_argument('-t', '--types', default='', help='List of types to convert the mx file specified by the -s flag.  Default: all')
```
This should be independent of -s 
This way someone can build all of the mx shaders but only for a specified subset of types.

amartinez-cg replied:
good thinking.  Added to the commit.

derekhaase commented on src/build-scripts/build_materialX_osl.py:
```diff
@@ -210,9 +210,13 @@ def main():
     parser.add_argument('-oslc_path', '--oslc_path', default='', help='Path to oslc executable.  Default: environment default')
     parser.add_argument('-compile', '--compile', default=0, help='Compile generated osl files 0|1.  Default: 0')
-    
+    parser.add_argument('-s', '--shader', default='', help='Specify a single mx shader to convert without the file extension, e.g. mx_add.  Default: none')
```
Maybe this could be a comma separated list of shaders to build.

amartinez-cg committed:
Adds flags to generate OSL only for specified mx files and types

Fixed help message to specify that the type list needs to be in ''

Made -t flag independant of -s flag, can now convert all shaders of one type

derekhaase commented on src/build-scripts/build_materialX_osl.py:
```diff
@@ -186,21 +186,29 @@ def mx_to_osl(shader, shader_types, options):
-            if options['compile']:
-                oso_filename = '%s.oso'%(osl_shadername)
-                osl_filepath = '%s.osl' % (osl_shadername)
+            if var_type in SHADER_TYPES:
```
might be more clear to rename shader_types or SHADER_TYPES so you don't have two variables with the same name.  
shader_types --> types_to_build

amartinez-cg replied:
Doh!  amateur hour....

amartinez-cg committed:
Adds flags to generate OSL only for specified mx files and types

Fixed help message to specify that the type list needs to be in ''

Made -t flag independant of -s flag, can now convert all shaders of one type

reworked code for -s and -t to be a little simpler

renames shader_types, more accurate count
